love_letter_sim2.py

```diff
@@ -130,15 +131,18 @@ def more_than_one_player(self):
 
     def take_turn(self):
         player = next(self.turn_order)
+        player.protected = False # effect of a previously played Handmaid expires
         print("START TURN", player.id)
         if player.eliminated is False:
             # JHA - seems to me that the action a prince would have would be
             # to call 'discard()' on other player.  That player would be
             # responsible for playing card, and if princess: elminated =
             # True, else draw_card()
+            # GM: I agree, if two princes are played on the same player one after the other, the current check below is "too late".
             if len(player.hand) is 0:  # deal with Prince
                 player.draw_a_card(self.deck)
```

Comment:
I still ended up taking this out.  If a prince is played against you, you discard what you have and, unless you're out, immediately draw a new card.  Waiting until now changes the deck order from the real game.